### DISCUSSION TITLE
How to parse embedded languages?

### DISCUSSION THREAD

**Discussion by pengsongkun741** [danipen/TextMateSharp#6] [Q&A] [answered]
I am trying to parse the [vue file](https://github.com/vuejs/vetur/tree/master/syntaxes) .
It is found that the embedded language inside cannot be parsed normally. What configuration is needed to achieve this goal?

In your sample code I see that the `GetFilePath(string scopeName)` method is automatically called in `IRegistryOptions`.
I can see that some embedded languages ​​needed by vue will find its tmlanguage files in this way. Is this related to the problem I mentioned?

**Comment by danipen** [comment, marked as answer]
Checkout the RegistryOptions provided for AvaloniaEdit. This may help. It supports loading dependent grammars.

https://github.com/AvaloniaUI/AvaloniaEdit/blob/feature/textmate-support/src/AvaloniaEdit.TextMate/RegistryOptions.cs
